src/argilla/client/feedback/training/base.py

```diff
@@ -347,8 +345,27 @@ def push_to_huggingface(self, repo_id: str, generate_card: Optional[bool] = True
 
             model_card.push_to_hub(repo_id, repo_type="model", token=kwargs["token"])
 
-
-class ArgillaTrainerSkeleton(ABC):
+    def update_config(self, *args, **kwargs) -> None:
+        # TODO: allow for backwards compatability for old methods by unpacking the args and kwargs from Pydantic models
+        new_args = []
+        for arg in args:
+            if hasattr(arg, "__fields__"):
+                new_args.extend(arg.dict().values())
+            else:
+                new_args.append(arg)
+        args = tuple(new_args)
+
+        new_kwargs = {}
+        for key, value in kwargs.items():
+            if hasattr(value, "__fields__"):
+                new_kwargs.update(value.dict())
+            else:
+                new_kwargs[key] = value
+        kwargs = new_kwargs
+        return super().update_config(*args, **kwargs)
```

Comment:
Just a reminder here. When we migrate to [pydantic 2](https://docs.pydantic.dev/latest/migration/#changes-to-pydanticbasemodel) we should update this method.